setup-tdx-host.sh

```diff
@@ -5,6 +5,21 @@ _error() {
   exit 1
 }
 
+KERNEL_VERSION=6.5.0.1003.3
+
+# grub: switch to kernel version
+grub_switch_kernel() {
+    KERNELVER=$1
+    MID=$(awk '/Advanced options for Ubuntu/{print $(NF-1)}' /boot/grub/grub.cfg | cut -d\' -f2)
+    KID=$(awk "/with Linux $KERNELVER/"'{print $(NF-1)}' /boot/grub/grub.cfg | cut -d\' -f2 | head -n1)
```

Comment:
On systems I've checked these lines won't match with the KERNEL_VERSION shown above. The existing grub.cfg on these systems shows:

```
menuentry 'Ubuntu, with Linux 6.5.0-1003-intel-opt'...
```

Reply:
The text you shown is not the version for apt

Reply:
Right, but `awk` won't match the appropriate line from `grub.cfg` because it uses a different version string compared to the version from `apt` (KERNELVER is using the `apt` version string). Unless I'm missing something.

Reply:
Ah, you are right, let me check